Compilation error: "'s' cannot be implicitly captured because no default capture mode has been specified"

https://github.com/rspatial/terra/blob/40992cf0fa56ae1538d046ab14083a8ca354f2e9/src/spatFactor.cpp#L26

Our tools have surfaced this compilation failure:

```
../src/spatFactor.cpp:25:36: error: no member named 'back_inserter' in namespace 'std'
                 std::end(v), std::back_inserter(result),
                              ~~~~~^
./src/spatFactor.cpp:28:22: error: variable 's' cannot be implicitly captured in a lambda with no capture-default specified
                     s.erase( s.find_last_not_of('0') + 1, std::string::npos );
                     ^
./src/spatFactor.cpp:27:33: note: 's' declared here
                    std::string s = std::to_string(d);
                                ^
./src/spatFactor.cpp:26:18: note: lambda expression begins here
                 [](T d) {
                 ^
./src/spatFactor.cpp:26:19: note: capture 's' by value
                 [](T d) {
                  ^
                  s
./src/spatFactor.cpp:26:19: note: capture 's' by reference
                 [](T d) {
                  ^
                  &s
./src/spatFactor.cpp:26:19: note: default capture by value
                 [](T d) {
                  ^
                  =
./src/spatFactor.cpp:26:19: note: default capture by reference
                 [](T d) {
                  ^
                  &
./src/spatFactor.cpp:28:31: error: variable 's' cannot be implicitly captured in a lambda with no capture-default specified
                     s.erase( s.find_last_not_of('0') + 1, std::string::npos );
                              ^
./src/spatFactor.cpp:27:33: note: 's' declared here
                    std::string s = std::to_string(d);
                                ^
./src/spatFactor.cpp:26:18: note: lambda expression begins here
                 [](T d) {
                 ^
./src/spatFactor.cpp:26:19: note: capture 's' by value
                 [](T d) {
                  ^
                  s
./src/spatFactor.cpp:26:19: note: capture 's' by reference
                 [](T d) {
                  ^
                  &s
./src/spatFactor.cpp:26:19: note: default capture by value
                 [](T d) {
                  ^
                  =
./src/spatFactor.cpp:26:19: note: default capture by reference
                 [](T d) {
                  ^
                  &
./src/spatFactor.cpp:29:22: error: variable 's' cannot be implicitly captured in a lambda with no capture-default specified
                     s.erase( s.find_last_not_of('.') + 1, std::string::npos );
                     ^
./src/spatFactor.cpp:27:33: note: 's' declared here
                    std::string s = std::to_string(d);
                                ^
./src/spatFactor.cpp:26:18: note: lambda expression begins here
                 [](T d) {
                 ^
./src/spatFactor.cpp:26:19: note: capture 's' by value
                 [](T d) {
                  ^
                  s
./src/spatFactor.cpp:26:19: note: capture 's' by reference
                 [](T d) {
                  ^
                  &s
./src/spatFactor.cpp:26:19: note: default capture by value
                 [](T d) {
                  ^
                  =
./src/spatFactor.cpp:26:19: note: default capture by reference
                 [](T d) {
                  ^
                  &
./src/spatFactor.cpp:29:31: error: variable 's' cannot be implicitly captured in a lambda with no capture-default specified
                     s.erase( s.find_last_not_of('.') + 1, std::string::npos );
                              ^
./src/spatFactor.cpp:27:33: note: 's' declared here
                    std::string s = std::to_string(d);
                                ^
./src/spatFactor.cpp:26:18: note: lambda expression begins here
                 [](T d) {
                 ^
./src/spatFactor.cpp:26:19: note: capture 's' by value
                 [](T d) {
                  ^
                  s
./src/spatFactor.cpp:26:19: note: capture 's' by reference
                 [](T d) {
                  ^
                  &s
./src/spatFactor.cpp:26:19: note: default capture by value
                 [](T d) {
                  ^
                  =
./src/spatFactor.cpp:26:19: note: default capture by reference
                 [](T d) {
                  ^
                  &
./src/spatFactor.cpp:30:29: error: variable 's' cannot be implicitly captured in a lambda with no capture-default specified
                     return s;
                            ^
./src/spatFactor.cpp:27:33: note: 's' declared here
                    std::string s = std::to_string(d);
                                ^
./src/spatFactor.cpp:26:18: note: lambda expression begins here
                 [](T d) {
                 ^
./src/spatFactor.cpp:26:19: note: capture 's' by value
                 [](T d) {
                  ^
                  s
./src/spatFactor.cpp:26:19: note: capture 's' by reference
                 [](T d) {
                  ^
                  &s
./src/spatFactor.cpp:26:19: note: default capture by value
                 [](T d) {
                  ^
                  =
./src/spatFactor.cpp:26:19: note: default capture by reference
                 [](T d) {
                  ^
                  &
```

I believe the highlighted line is the issue; per my read of this SO Q&A and a cursory understanding of the code, I believe `[&s]` is the correct usage:

https://stackoverflow.com/q/30217956/3576984

--- a/src/spatFactor.cpp
+++ b/src/spatFactor.cpp
@@ -1,3 +1,4 @@
+#include <iterator>
 #include <vector>
 #include <string>
 #include <numeric>

